kubetest/kind.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// This deployer requires --root=/go/src when pulling repositories.
+const (
+	kindBinaryPath = "/go/bin"
+	kindSrcPath    = "/go/src/sigs.k8s.io/kind"
+	k8sSrcPath     = "/go/src/k8s.io/kubernetes"
+	kindNodeImage  = "kindest/node:latest"
+)
+
+var (
+	kindConfigPath = flag.String("kind-config-path", "",
+		"(kind only) Path to the kind configuration file.")
+)
+
+// Definition for the kind deployer.
+type kind struct {
+	configPath string
+}
+
+// newKind creates a new kind deployer.
+func newKind() (deployer, error) {
+	k, err := initializeKind()
+	if err != nil {
+		return nil, err
+	}
+	return k, nil
+}
+
+// initializeKind initializers the kind deployer flags.
+func initializeKind() (*kind, error) {
+	// Ensure the kind binary path is added in $PATH.
+	// Could be redundant if kindBinaryPath is /go/bin and this is already in $PATH.
+	path := os.Getenv("PATH")
```

Comment:
I think this is fine for the MVP, but as a future thought (this has tradeoffs, just want to discuss the idea, not convinced one way or the other!):

I wonder if instead we should use LookPath() to look for kind, and keep a copy of the full path by which to invoke kind in the deployer 🤔 

Upside to this one: a test_cmd can easily invoke kind (though we could populate the sub command env instead..)

Reply:
not sure what is best here for now.